azurerm/internal/services/mssqlvm/resource_arm_mssql_virtual_machine.go

```diff
@@ -0,0 +1,559 @@
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+						"sql_log_luns": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeInt,
+							},
+						},
+						"sql_temp_db_default_file_path": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+						"sql_temp_db_luns": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeInt,
+							},
+						},
+						"storage_workload_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(sqlvirtualmachine.GENERAL),
+								string(sqlvirtualmachine.OLTP),
+								string(sqlvirtualmachine.DW),
+							}, false),
+						},
+					},
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmMsSqlVirtualMachineCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MSSQLVM.SQLVirtualMachinesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+
+	id, _ := azure.ParseAzureResourceID(d.Get("virtual_machine_resource_id").(string))
+	name := id.Path["virtualMachines"]
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroupName, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for present of existing Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q): %+v", name, resourceGroupName, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_sql_virtual_machine", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	sqlServerLicenseType := d.Get("sql_server_license_type").(string)
+	virtualMachineResourceName := d.Get("virtual_machine_resource_id").(string)
+
+	properties := sqlvirtualmachine.Properties{
+		SQLServerLicenseType:     sqlvirtualmachine.SQLServerLicenseType(sqlServerLicenseType),
+		VirtualMachineResourceID: &virtualMachineResourceName,
+	}
+
+	if sqlVirtualMachineGroupResourceID, ok := d.GetOk("sql_virtual_machine_group_resource_id"); ok {
+		SQLVirtualMachineGroupResourceID := sqlVirtualMachineGroupResourceID.(string)
+		properties.SQLVirtualMachineGroupResourceID = &SQLVirtualMachineGroupResourceID
+	}
+
+	if sqlImageSku, ok := d.GetOk("sql_image_sku"); ok {
+		SQLImageSku := sqlvirtualmachine.SQLImageSku(sqlImageSku.(string))
+		properties.SQLImageSku = SQLImageSku
+	}
+
+	if _, ok := d.GetOk("auto_patching_settings"); ok {
+		properties.AutoPatchingSettings = expandArmSqlVirtualMachineAutoPatchingSettings(d)
+	}
```

Comment:
We should just move the check into the function:
```
properties.AutoPatchingSettings = expandArmSqlVirtualMachineAutoPatchingSettings(d.Get("auto_patching_settings"))
```

and then the function can check if its nil & contains any elements